start_admin.sh

```diff
@@ -69,14 +76,51 @@ EOF
   chmod 644 "${proxy_profile}"
 }
 
+# Set up agetty services for serial console access and the sshd daemon service
+enable_systemd_services() {
+  # Grab `console=` parameters from the kernel command line.
+  CONSOLES=()
+  for opt in $(cat /proc/cmdline) ; do
+     optarg="$(expr "${opt}" : '[^=]*=\(.*\)' ||:)"
+     case "${opt}" in
+        console=*) CONSOLES+=("${optarg%,*}") ;;
+     esac
+  done
+
+  HOST_DEVTMPFS="/.bottlerocket/rootfs/dev"
+  for console in "${CONSOLES[@]}" ; do
+     # Skip devices that don't exist.
+     [ -c "${HOST_DEVTMPFS}/${console}" ] || continue
+
+     # Otherwise instantiate a service from the template unit. This is normally
+     # done by `systemd-getty-generator`, but that skips over ordinary devices
+     # when run inside a container.
+     case "${console}" in
+        ttyS*|ttyAMA*|ttyUSB*)
+           systemctl --user enable "serial-getty@${console}.service"
+           ;;
+        tty*)
+           systemctl --user enable "getty@${console}.service"
+           ;;
+     esac
+  done
+  # Enable the SSH daemon service unit so we can run it in the background
+  systemctl --user enable "sshd.service"
+}
+
 # Create local user
 echo "${LOCAL_USER} ALL=(ALL) NOPASSWD: ALL" > "/etc/sudoers.d/${LOCAL_USER}"
 chmod 440 "/etc/sudoers.d/${LOCAL_USER}"
 useradd -m -G users,api "${LOCAL_USER}"
+if [ -z "${PASSWORD_HASH}" ]; then
+  passwd -d "${LOCAL_USER}"
```

Comment:
We want to disable password logins in that case (`usermod -L "${LOCAL_USER}"`, assuming that still allows ssh) to avoid exposing a password-less serial console on upgrade. It's not a given that someone with serial console access should have the ability to log into a privileged account on the host.